lib/auth.ts

```diff
@@ -143,7 +143,7 @@ export const config: NextAuthConfig = {
           linkedAccounts[provider] = profile?.login as string
           break;
         case 'twitter':
-          linkedAccounts[provider] = (profile?.data as any)?.username as string
+          linkedAccounts[provider] = (profile?.data as any)?.id as string
```

Comment:
In case you want to add both the id and the user handle, an easy way would be to concatenating the string with a certain character. For example:

```
linkedAccounts[provider] = `${(profile?.data as any)?.id}:${(profile?.data as any)?.username}`
```

However, you would also need to handle splitting back in the contract or whenever you need access to either of the fields